code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -135,6 +135,22 @@ var/const/VOX_PATH = "sound/vox_fem/"
 	for(var/word in words)
 		play_vox_word(word, src.z, null)
 
+	ai_voice_announcement_to_text(words, src)
+
+
+/proc/ai_voice_announcement_to_text(words, ai)
+	var/mob/living/silicon/ai/AI = ai
+	var/words_string = jointext(words, " ")
+	var/formatted_message = "<h1 class='alert'>A.I. Announcement</h1>"
+	formatted_message += "<br><span class='alert'>[words_string]</span>"
+	formatted_message += "<br><span class='alert'> -[AI]</span>"
+
+	for(var/mob/M in GLOB.player_list)

```

Comment:
Can use `var/thingy` and then cast it to a mob in the loop seeing the list should only ever contain mobs.

Reply:
If you know they should all be mobs, wouldn't this be more efficient, with not having to cast it? Or can it lead to some fucky behavior?

Reply:
I believe if you specify a type in the for loop that it will filter out the other types. A filter costs time linear to the list length. So if you already know it only has only one type of variable (or it's subtypes) then you can better use `var/thingy` and cast it to it's actual type in the loop itself

Reply:
Fair enough